phalcon/DataMapper/Query/AbstractConditions.zep

```diff
@@ -0,0 +1,358 @@
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalcon.io>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ *
+ * Implementation of this file has been influenced by AtlasPHP
+ *
+ * @link    https://github.com/atlasphp/Atlas.Query
+ * @license https://github.com/atlasphp/Atlas.Qyert/blob/1.x/LICENSE.md
+ */
+
+namespace Phalcon\DataMapper\Query;
+
+use Phalcon\Helper\Arr;
+
+/**
+ * Class AbstractConditions
+ */
+abstract class AbstractConditions extends AbstractQuery
+{
+    /**
+     * Sets the `LIMIT` clause
+     *
+     * @param int $limit
+     *
+     * @return AbstractConditions
+     */
+    public function limit(int limit) -> <AbstractConditions>
+    {
+        let this->store["LIMIT"] = limit;
```

Comment:
I'm wondering if it would be wise to refactor store to a class so we can have proper checking on the keys.

Reply:
I went with the fastest approach i.e. array vs object with an array etc. We can change that for sure. Up to you.

Reply:
I got a feeling moving this to a class will help us to maintain this in the future.  Will the performance impact be big?